edge-util/src/Microsoft.Azure.Devices.Edge.Util/Option.cs

```diff
@@ -211,7 +216,6 @@ public static class Option
         /// </summary>
         public static Option<T> None<T>() => new Option<T>(default(T), false);
 
-        public static Option<T> Maybe<T>(T value) where T : class =>
-            value == null ? None<T>() : Some(value);
+        public static Option<T> Maybe<T>(T value) => value == null ? None<T>() : Some(value);
```

Comment:
It feels misleading to allow `Maybe` for value-types. The error from the original code would've incentivized the programmer to use `Some()` instead of `Maybe()`.

Reply:
Yes.  Basically we should always use Some or None; this method is recently added by Angelo; will remove it in a separate PR.

Reply:
What? Why would you remove it? It's useful for converting from nullable `T` to `Option<T>` ?

Reply:
Basically current usages of Maybe can be avoided; by calling Some or None earlier; tha's why we don't need it previously.  I am ok to keep this method, but it will be only useful for any optional argument passing from external system only.  And currently I don't see many of this case.

Reply:
Thanks for your comment.  Will revert back this change.